cpp/src/barretenberg/ecc/curves/bn254/g1.hpp

```diff
@@ -17,4 +17,11 @@ struct Bn254G1Params {
 };
 
 typedef group<fq, fr, Bn254G1Params> g1;
-} // namespace barretenberg
\ No newline at end of file
+
+} // namespace barretenberg
+
+// specialize the name in msgpack schema generation
+inline std::string msgpack_schema_name(barretenberg::g1::affine_element const&)
+{
+    return "G1AffineElement";
+}
```

Comment:
It is largely related to C++ templates. Recall that this would expand to some affine_element<MetaStruct<Parameters>>. It's quite hard to parse that on the TS side, and then if any refactor were done, we would have to do it again. It felt more stable to put the maintenance burden on the coder. I didn't know how much to capture of that in the comment